airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/backward_compatibility.py

```diff
@@ -227,5 +227,6 @@ def check_if_stream_was_removed(self, diff: DeepDiff):
 
     def check_if_cursor_field_was_changed(self, diff: DeepDiff):
         """Check if a default cursor field value was changed."""
-        if diff:
+        invalid_changes = {"values_changed", "iterable_item_added", "iterable_item_removed"}
+        if diff and set(diff.keys()).issubset(invalid_changes):
```

Comment:
could this now pass in a case where we have a new stream _but also_ breaking cursor field change on another stream?

Reply:
Thank for the suggestion, I added a corresponding test case and discovered the check was not robust enough: improved in 3231e6d41e6c0690139feb977520178774fe1e17